core/src/main/java/feign/Request.java

```diff
@@ -187,7 +189,9 @@ public Charset charset() {
    * @see #charset()
    */
   public byte[] body() {
-    return body.data;
+    return (body == null)
+        ? Body.EMPTY.data
```

Comment:
FWIW, I hate all the `null`s and mutable code we have laying around (specially on classes that mix mutable an immutable code)

That is probably an awesome, compatibility breaking clean up for feign 11.

Reply:
> any particular reason to not return `null`?

Well, data is on object so somebody could call body().length and face NullPointerException. If we provide empty array - we'll avoid such problem

Reply:
We could add an `isEmpty` function instead to make our intentions clearer.  Thoughts?

Reply:
For Body? Honestly I didn't get the idea clearly. Could you provide a bit more info please?

Reply:
We can add a method to `Request` that indicates if the request is "Empty" or not, instead of relying on `null` checks.  For this particular issue we can leave the current implementation alone, but add an additional method, such as `hasBody` to deal with the state check.

```java
/**
 * The Request body.  Can be {@code null}
 *
 * @returns the Request body as a byte array. 
 */
public byte[] body() {
   return body != null ? body.data : null;
}

/**
 * Returns if this Request has a Body defined.
 *
 * @returns {@code true} if this request has a body defined, {@code false} otherwise.
 */
public boolean hasBody() {
   return body != null ? body.data != null : false;
}
```

This type of change accomplishes two things.  First, it doesn't break compatibility for older clients.  Second it creates a pattern that encourages users to check first before use.  In a breaking change, we could update the `body` method to return an `Optional` to be even more explicit.

Reply:
Thank you, @kdavisk6 
Got it now